server/src/main/java/org/apache/iotdb/db/query/reader/seriesRelated/NewSeriesReaderWithoutValueFilter.java

```diff
@@ -52,22 +61,44 @@
 
   private PageHeader cachedPageHeader;
   private boolean hasCachedNextPage;
-
-  public NewSeriesReaderWithoutValueFilter(Path seriesPath, TSDataType dataType, Filter timeFilter,
-      QueryContext context) throws StorageEngineException, IOException {
-    QueryDataSource queryDataSource = QueryResourceManager.getInstance()
-        .getQueryDataSource(seriesPath, context);
-    timeFilter = queryDataSource.setTTL(timeFilter);
+  private IAggregateReader overlappedChunkReader;
+  private long latestDirectlyOverlappedPageEndTime;
+
+  public NewSeriesReaderWithoutValueFilter(QueryDataSource queryDataSource, TSDataType dataType,
+      Filter timeFilter, QueryContext context) throws IOException {
+    Path seriesPath = queryDataSource.getSeriesPath();
+    TreeSet<TsFileResource> unseqTsFilesSet = new TreeSet<>((o1, o2) -> {
+      String queryMeasurement = seriesPath.getMeasurement();
+      List<Long> o1StartTimeList = o1.getChunkMetaDataList().stream()
```

Comment:
This ChunkMetadataList should all belongs to this measurement. No need to filter again.

The sort could compare by the startTimeMap

Reply:
done